src/briefcase/console.py

```diff
@@ -124,6 +126,11 @@ def __init__(self, enabled=True):
         self.rich_console = rich_console
         self._enabled = enabled
 
+        # Signal that Rich is dynamically controlling the terminal output.
+        # Therefore, all output must be printed to the screen by Rich to
+        # prevent corruption of dynamic elements like Wait Bars.
+        self.is_output_controlled = False
```

Comment:
I decided to generalize the indication and handling of an active Wait Bar in to any active Rich content. However, "there are two hard things in computer science: cache invalidation and naming things". I struggled with a name for this flag that communicated what its actually indicating. Open to a better name.